src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/camunda/payment/PaymentsCallbackHandler.java

```diff
@@ -57,10 +60,23 @@ private CallbackResponse makePbaPayment(CallbackParams callbackParams) {
         List<String> errors = new ArrayList<>();
         try {
             var paymentReference = paymentsService.createCreditAccountPayment(caseData, authToken).getReference();
-            caseData = caseData.toBuilder()
-                .paymentDetails(PaymentDetails.builder().status(SUCCESS).reference(paymentReference).build())
+            CaseData.CaseDataBuilder builder = caseData.toBuilder();
+            PaymentDetails paymentDetails = ofNullable(caseData.getClaimIssuedPaymentDetails())
+                .map(PaymentDetails::toBuilder).orElse(PaymentDetails.builder())
+                .status(SUCCESS)
+                .reference(paymentReference)
+                .build();
+
+            if (featureToggleService.isFeatureEnabled("payment-reference")) {
```

Comment:
at what point are we going to toggle the feature on?

Reply:
so the test env is currently set up with this toggled off, but this probably isn't enough as this counts for local, demo, aat etc. I guess this is a tough one. It should be turned on as soon as the definition repo PR is ready to be merged, but doing so will actually be a breaking change temporarily 😕